qiskit/assembler/assemble_schedules.py

```diff
@@ -13,213 +13,222 @@
 # that they have been altered from the originals.
 
 """Assemble function for converting a list of circuits into a qobj."""
+from typing import Any, Dict, List, Tuple
 from collections import defaultdict
 
 from qiskit.exceptions import QiskitError
-from qiskit.pulse.commands import (PulseInstruction, AcquireInstruction,
+from qiskit.pulse import Schedule
+from qiskit.pulse.commands import (Command, PulseInstruction, Acquire, AcquireInstruction,
                                    DelayInstruction, SamplePulse, ParametricInstruction)
-from qiskit.qobj import (PulseQobj, QobjExperimentHeader,
+from qiskit.qobj import (PulseQobj, QobjHeader, QobjExperimentHeader,
                          PulseQobjInstruction, PulseQobjExperimentConfig,
                          PulseQobjExperiment, PulseQobjConfig, PulseLibraryItem)
 from qiskit.qobj.converters import InstructionToQobjConverter, LoConfigConverter
 from qiskit.qobj.converters.pulse_instruction import ParametricPulseShapes
 from qiskit.qobj.utils import MeasLevel, MeasReturnType
 
+from .run_config import RunConfig
 
-def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
+
+def assemble_schedules(schedules: List[Schedule],
+                       qobj_id: int,
+                       qobj_header: QobjHeader,
+                       run_config: RunConfig) -> PulseQobj:
     """Assembles a list of schedules into a qobj that can be run on the backend.
 
     Args:
-        schedules (list[Schedule]): schedules to assemble
-        qobj_id (int): identifier for the generated qobj
-        qobj_header (QobjHeader): header to pass to the results
-        run_config (RunConfig): configuration of the runtime environment
+        schedules: schedules to assemble
+        qobj_id: identifier for the generated qobj
+        qobj_header: header to pass to the results
+        run_config: configuration of the runtime environment
+
     Returns:
-        PulseQobj: the Qobj to be run on the backends
+        The Qobj to be run on the backends
+
     Raises:
-        QiskitError: when invalid schedules or configs are provided
+        QiskitError: when LO frequency settings are not supplied
     """
-    if hasattr(run_config, 'instruction_converter'):
-        instruction_converter = run_config.instruction_converter
-    else:
-        instruction_converter = InstructionToQobjConverter
-
-    qobj_config = run_config.to_dict()
-
-    qubit_lo_freq = qobj_config.get('qubit_lo_freq', None)
-    if qubit_lo_freq is None:
+    if not hasattr(run_config, 'qubit_lo_freq'):
```

Comment:
yes, if a `backend` isn't supplied to `assemble`, since the user has the option to supply all the different required arguments rather than a `backend`

Reply:
Oh, I see now why you asked though.

I just tried this in a jupyter notebook:
```
assemble(sch, qubit_lo_freq=None)
```
and I still got the error `QiskitError: 'qubit_lo_freq must be supplied.'` with this branch checked out.

I wasn't too crazy about this part either, because it validates one particular thing while leaving so much unvalidated. The user could set `qubit_lo_freq="str"` and avoid hitting this line 🤷‍♀